### DISCUSSION TITLE
build virtual camera path based on original camera positions.

### DISCUSSION THREAD

**Discussion by felixgeen** [NVlabs/instant-ngp#506] [Q&A]
Thank you for this amazing tool! I am enjoying it so much. I want to render some camera flythroughs based on my original capture cameras path. Is there some way to turn the transforms.json that colmap creates into a format similar to base_cam.json so that I can make a flythrough video which matches the same path my capture camera took?
![image](https://user-images.githubusercontent.com/32222984/164245524-a79f9102-0232-4eb7-859e-2506ae0bf296.png)

**Comment by Tom94** [comment]
While __instant-ngp__ does not natively have such a conversion tool, there are two slightly more involved approaches to achieve what you want:
1. __Programmatic:__ if you're feeling adventurous, you could convert the coordinate system of the `transforms.json`'s camera matrices into NGP coordinates (see diagrams and discussion here: https://github.com/NVlabs/instant-ngp/discussions/153?converting=1) and then create the rotation quaternion `Q` and translation vector `T` from these. (See `camera_path.cu` for other members each keyframe should have.)
2. __Manual:__ the testbed GUI allows you to snap the camera to the training views in sequence. You could manually step through the sequence and add keyframes to the camera path. Due to the camera path's smooth interpolation, I expect you can get away with skipping many training views inbetween steps and still produce a reasonable path.

Cheers!

**Reply by charlesxwang** [reply]
@Tom94 Can you give a demo camera path json file, say for the fox example?
